test/libsyncengine/syncpal/testsyncpal.h

```diff
@@ -25,26 +26,31 @@ namespace KDC {
 
 class TestSyncPal : public CppUnit::TestFixture {
         CPPUNIT_TEST_SUITE(TestSyncPal);
-        CPPUNIT_TEST(testAll);
-        CPPUNIT_TEST(testConflictQueue);
+        CPPUNIT_TEST(testUpdateTree);
+        CPPUNIT_TEST(testSnapshot);
+        CPPUNIT_TEST(testCopySnapshots);
+        CPPUNIT_TEST(testOperationSet);
         CPPUNIT_TEST_SUITE_END();
 
     public:
         void setUp() override;
         void tearDown() override;
 
-    protected:
-        void testAll();
-        void testConflictQueue();
-
     private:
         std::shared_ptr<SyncPal> _syncPal;
         std::shared_ptr<ParmsDb> _parmsDb;
-
+        std::shared_ptr<LocalTemporaryDirectory> _localTempDir;
```

Comment:
Why is it better to hold a shared pointer instead of an object?

Reply:
I didn't realize that there was a risk of temp dir name collisions across tests (we use the time stamp, but the truncation is around 1 minute). I believe we should make the temporary directory name unique for each test. Otherwise some tests may share the same temporary directory, with unpredictable side effects.

Reply:
Following our discussion, a new instance of the testFixture is indeed created for each test. I have removed the use of shared_ptr and improved the behavior of localTemporaryDirectory to ensure that two tests do not use the same directory. [7d95284](https://github.com/Infomaniak/desktop-kDrive/pull/203/commits/7d952842d3a24c69e60e9330d937db0149bb9ca8)